dockstore-webservice/src/test/java/io/dockstore/webservice/TokenResourceTest.java

```diff
@@ -0,0 +1,198 @@
+    private static final DockstoreWebserviceConfiguration configuration = mock(DockstoreWebserviceConfiguration.class);
+    // response.readEntity(Token.class) appears to have issues reading the "token" property, using a different mapper instead
+    private static ObjectMapper mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    ;
+    private static TokenResource tokenResourceSpy = spy(
+            new TokenResource(tokenDAO, enduserDAO, client, cachingAuthenticator, configuration));
+
+    @ClassRule
+    public static final ResourceTestRule resources = ResourceTestRule.builder().addResource(tokenResourceSpy)
+            .addProvider(new AuthValueFactoryProvider.Binder<>(User.class)).build();
+
+    private static TokenResponse getFakeTokenResponse() {
+        TokenResponse fakeTokenResponse = new TokenResponse();
+        fakeTokenResponse.setAccessToken("fakeAccessToken");
+        fakeTokenResponse.setExpiresInSeconds(9001l);
+        fakeTokenResponse.setRefreshToken("fakeRefreshToken");
+        return fakeTokenResponse;
+    }
+
+    private static Userinfoplus getFakeUserinfoplus() {
+        Userinfoplus fakeUserinfoplus = new Userinfoplus();
+        fakeUserinfoplus.setEmail("potato@gmail.com");
+        fakeUserinfoplus.setGivenName("Beef");
+        fakeUserinfoplus.setFamilyName("Stew");
+        fakeUserinfoplus.setGender("New classification");
+        fakeUserinfoplus.setPicture("https://i.kym-cdn.com/photos/images/newsfeed/001/207/210/b22.jpg");
+        return fakeUserinfoplus;
+    }
+
+    private static Token getFakeToken() {
+        Token fakeToken = new Token();
+        fakeToken.setContent("fakeContent");
+        fakeToken.setTokenSource(TokenType.DOCKSTORE);
+        fakeToken.setUserId(1);
+        fakeToken.setUsername("fakeTokenUsername");
+        return fakeToken;
+    }
+
+    private static User getFakeUser() {
+        User fakeUser = new User();
+        fakeUser.setUsername("fakeUser");
+        fakeUser.setIsAdmin(false);
+        fakeUser.setAvatarUrl("https://i.kym-cdn.com/photos/images/newsfeed/001/207/210/b22.jpg");
+        fakeUser.setCurator(false);
+        return fakeUser;
+    }
+
+    @Before
+    public void setup() {
+        doReturn(getFakeTokenResponse()).when(tokenResourceSpy).getTokenResponse(anyString(), anyString());
+        doReturn(getFakeUserinfoplus()).when(tokenResourceSpy).getUserInfo(anyString());
+        doReturn(1l).when(tokenDAO).create(any());
+        doReturn(getFakeToken()).when(tokenDAO).findById(any());
+        doReturn(getFakeUser()).when(enduserDAO).findById(anyLong());
+    }
+
+    @After
+    public void tearDown() {
+        reset(tokenDAO);
+        reset(enduserDAO);
+        reset(tokenResourceSpy);
+    }
+
+    /**
+     * For a non-existing user, checks that two tokens (Dockstore and Google) were created
```

Comment:
This checks on a dockstore token, it doesn't seem to check on a google token?

Reply:
Checks that two tokens were created. Checks that the Dockstore token is returned (GitHub endpoint also returns the Dockstore token)

Reply:
Ah ok, please check that one of each is created